### ISSUE TITLE
Using license_specifications into terraform EKS self_managed_node_group module

### ISSUE THREAD

**Issue by Lamaspanzer** [terraform-aws-modules/terraform-aws-eks#2753]
## Description

I'm encountering an issue while attempting to define license_specifications within my Terraform configuration. 
The error message I'm getting `A managed resource "license_specifications" "value" has not been declared in module.self_managed_node_group`

## Versions

Terraform v1.5.6 on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v5.15.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.23.0

+ module AWS EKS version 19.16.0

## Code

```tf
module "self_managed_node_group" {
  source  = "terraform-aws-modules/eks/aws//modules/self-managed-node-group"
  version = "19.16.0"

  name                = local.eks_name
  cluster_name        = module.eks.cluster_name
  cluster_version     = var.eks_version
  cluster_endpoint    = module.eks.cluster_endpoint
  cluster_auth_base64 = module.eks.cluster_certificate_authority_data

  ....

  placement = {
    tenancy                 = "host"
    host_resource_group_arn = "arn:aws:resource-groups:us-east-1:XXXXXXX:group/XXXXXX"
  }

  license_specifications = {
    license_configuration_arn = "arn:aws:license-manager:us-east-1:XXXXXX:license-configuration:XXXXXXXXX"
  }
}
```

### Output

```tf
 Error: Reference to undeclared resource
│ 
│   on .terraform/modules/self_managed_node_group/modules/self-managed-node-group/main.tf line 288, in resource "aws_launch_template" "this":
│  288:       license_configuration_arn = license_specifications.value.license_configuration_arn
│ 
│ .
```

## Additional context

It works by applying this modification to the dynamic block: 
```tf
  dynamic "license_specification" {
    for_each = length(var.license_specifications) > 0 ? var.license_specifications : []

    content {
      license_configuration_arn = license_specification.value.license_configuration_arn
    }
  }
```

conf
```tf
  license_specifications = [
    {
      license_configuration_arn = "arn:aws:license-manager:us-east-1:XXXXXX:license-configuration:lic-XXXXXXXXX"
    }
  ]
```
from 
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/918aa7cc40cbc072836410747834de64d84f514d/modules/self-managed-node-group/main.tf#L288

But this change probably isn't the best ide, it doesn't make sense to add the map to a list in this case.

**Comment by Lamaspanzer** [comment]
I think that the extra 's' in `license_configuration_arn = license_specificationS.value.license_configuration_arn`
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/918aa7cc40cbc072836410747834de64d84f514d/modules/self-managed-node-group/main.tf#L288
 is the cause of the `Reference to undeclared resource`

**Comment by bryantbiggs** [comment]
yep, that looks incorrect. feel free to open a PR and I'll merge it, thanks!

**Comment by antonbabenko** [comment]
This issue has been resolved in version 19.17.3 :tada:
